src/fstext/determinize-star-inl.h

```diff
@@ -468,99 +469,232 @@ template<class Arc> class DeterminizerStar {
+    // now we are going to propagate the "unprocessed_weight"
+    for (ArcIterator<Fst<Arc> > aiter(*ifst_, elem.state);
+         !aiter.Done(); aiter.Next()) {
+      const Arc &arc = aiter.Value();
+      if (sorted && arc.ilabel > 0) {
+        break;
+        // Break from the loop: due to sorting there will be no
+        // more transitions with epsilons as input labels.
+      }
+      if (arc.ilabel == 0) {  // Epsilon transition.
+        Element next_elem;
+        next_elem.state = arc.nextstate;
+        next_elem.weight = Weight::Zero();
+        Weight next_unprocessed_weight
+                       = Times(unprocessed_weight, arc.weight);
+
+        // now must append strings
+        if (arc.olabel == 0)
+          next_elem.string = elem.string;
+        else {
+          vector<Label> seq;
+          repository_.SeqOfId(elem.string, &seq);
+          if (arc.olabel != 0)
+            seq.push_back(arc.olabel);
+          next_elem.string = repository_.IdOfSeq(seq);
+        }
+        typename std::map<InputStateId, EpsilonClosureInfo>::iterator
+            iter = cur_subset->find(next_elem.state);
+        if (iter == cur_subset->end()) {
+          // was no such StateId: insert and add to queue.
+          (*cur_subset)[next_elem.state] =
+                       EpsilonClosureInfo(next_elem,
+                                          next_unprocessed_weight,
+                                          true,
+                                          this_depth + 1
+                                          );
+          queue.push(IdAndDepth(next_elem.state, this_depth + 1));
+        } else {  // one is already there.  Add weights.
+          if (iter->second.element.string != next_elem.string) {
+            std::cerr << "DeterminizerStar: FST was not functional "
+              "-> not determinizable\n";
+            { // Print some debugging information.  Can be helpful to debug
+              // the inputs when FSTs are mysteriously non-functional.
+              vector<Label> tmp_seq;
+              repository_.SeqOfId(iter->second.element.string, &tmp_seq);
+              std::cerr << "First string: ";
+              for (size_t i = 0; i < tmp_seq.size(); i++)
+                std::cerr << tmp_seq[i] << " ";
+              std::cerr << "\nSecond string: ";
+              repository_.SeqOfId(next_elem.string, &tmp_seq);
+              for (size_t i = 0; i < tmp_seq.size(); i++)
+                std::cerr << tmp_seq[i] << " ";
+              std::cerr << "\n";
+            }
+            throw std::runtime_error("Non-functional FST: cannot determinize.\n");
+          }
+          if (iter->second.in_queue) {
+            // since in queue, no check since we're propagating them anyway
+            iter->second.weight_to_process =
+                Plus(iter->second.weight_to_process, next_unprocessed_weight);
+            // probably should take it out of the queue, update the depth
```

Comment:
I am not sure how to do that... It seems to do that we need to pass to the priority_queue a comparison class; if we rely on the information in "info", it'll make the comparison class not "well defined", since it will keep changing?